protocol.txt

```diff
@@ -0,0 +1,124 @@
+        0 1 2 3 4 5 6 7 0 1 2 3 4 5 6 7 0 1 2 3 4 5 6 7 0 1 2 3 4 5 6 7
+       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+     0 |                                                               |
+       +                            Intron                             +
+     8 |                                                               |
+       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    12 |       6       |       0       |             Size              |
+       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    16 |                                                               |
+       +                          New intron                           +
+    20 |                                                               |
+       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    24 |                                                               |
+       ~                                                               ~
+       ~                         Client config                         ~
+       ~                                                               ~
+       |                                                               |
+       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+
+    New intron:     To prevent prediction attacks, printer sends a new
+                    randomly generated intron. This intron must be used
+                    in all subsequent communication.
+
+    Client config:  This field contains serialized SSID and password.
+                    Both of them are serialized as 1 byte Length
+                    followed by Length bytes.
+
+
+MSG_PACKET_V2
+    This message is sent by both printer and ESP.
+
+        0               1               2               3
+        0 1 2 3 4 5 6 7 0 1 2 3 4 5 6 7 0 1 2 3 4 5 6 7 0 1 2 3 4 5 6 7
+       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+     0 |                                                               |
+       +                            Intron                             +
+     8 |                                                               |
+       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    12 |       7       |      Up       |             Size              |
+       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    16 |                                                               |
+       ~                                                               ~
+       ~                            Packet                             ~
+       ~                                                               ~
+       |                                                               |
+       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+
+    Up:             This field is sent by printer to indicate
+                    connection state of the Wi-Fi.
```

Comment:
What does „Up“ mean in the printer -> ESP direction? Ignored?

Reply:
Yes, it is ignored when sent from printer.